ubi/src/extension.rs

```diff
@@ -108,7 +110,17 @@ fn extension_is_part_of_version(path: &str, ext_str: &OsStr) -> bool {
 }
 
 fn extension_is_platform(ext_str: &OsStr) -> bool {
-    ALL_OSES_RE.is_match(ext_str.to_string_lossy().as_ref())
+    static PLATFORM_RE: Lazy<Regex> = Lazy::new(|| {
+        Regex::new(
+            &[ALL_OSES_RE.as_str(), ALL_ARCHES_RE.as_str()]
+                .iter()
+                .map(|r| format!("({r})"))
```

Comment:
```suggestion
                .map(|r| format!("(?:{r})"))
```

Reply:
I'm going to add this to the os and arch regexes too. This PR just made me notice this was missing.